src/span_buffer.h

```diff
@@ -42,6 +58,32 @@ struct PendingTrace {
   std::string hostname;
   double analytics_rate;
   SampleResult sample_result;
+  // `trace_tags` are tags that are associated with the entire local trace,
+  // rather than with a single span.  Some other tags are added to the local
+  // root span when the trace chunk is sent to the agent (see
+  // `finish_root_span` in `span_buffer.cpp`).  In addition to those tags,
+  // `trace_tags` are similarly added.  `trace_tags` names all begin with the
+  // same prefix, `trace_tag_prefix`, defined in `tag_propagation.h`.
+  // `trace_tags` originate from extracted trace context (`SpanContext`).  Some
+  // trace tags require special handling, e.g. `upstream_services`, below.
+  std::unordered_map<std::string, std::string> trace_tags;
+  // `upstream_services` contains the parsed value of the
+  // "_dd.p.upstream_services" tag from the trace tags.  Additionally, if this
+  // service is the first to make a sampling decision, or if it changes the
+  // sampling decision (currently not possible), `upstream_services` ends with
+  // an `UpstreamService` object describing this service's sampling decision.
+  // `upstream_services` is attached to the local root span as the
+  // "_dd.p.upstream_services" tag.
+  std::vector<UpstreamService> upstream_services;
+  // `service` is the same service name specified in `TracerOptions` or deduced
+  // from `DD_SERVICE`.
+  std::string service;
```

Comment:
Caleb: When the service is set via a span tag, this variable has to be updated (like in the case of manual keep/drop affecting sampling priority). We cannot assume that the tracer config service is the service here.